api/submission/fetch_course_report.go

```diff
@@ -0,0 +1,26 @@
+package submission
+
+import (
+    "github.com/edulinq/autograder/api/core"
+    "github.com/edulinq/autograder/report"
+)
+
+type FetchCourseReportRequest struct {
+    core.APIRequestCourseUserContext
+    core.MinRoleGrader
+}
+
+type FetchCourseReportResponse struct {
+    CourseReport *report.CourseScoringReport `json:"course-report"`
+}
+
+func HandleFetchCourseReport(request *FetchCourseReportRequest) (*FetchCourseReportResponse, *core.APIError) {
+    gettingCourseReport, err := report.GetCourseScoringReport(request.Course);
+
```

Comment:
Remove this newline here.
We want to have the error check right next to the code that may produce the error.

As we go through this PR, you will notice comment like this (and the one above about variable names) that may seem pedantic and do not change how the code runs, but are actually very important from a software engineering viewpoint.
Things like this are important to make our code as readable (which helps towards maintainability) as possible.
Whenever you write code, try to think how someone who just read a few hundred lines of code will be able to read it.